tests/core/test_embind_5.out

```diff
@@ -3,3 +3,6 @@ doing it
 constructing my foo
 constructing my bar
 doing something else
+destructing my foo
+destructing my bar
+destructing my foo
```

Comment:
why is foo destroyed twice, but bar once? (i see just one creation of each, so i'd expect once for each)

Reply:
The destructor for MyBar also calls the destructor for MyFoo because MyBar is a subclass of MyFoo.

Reply:
Oh right... there's a programming interview question I would have gotten wrong...